feature/topic/src/main/java/com/google/samples/apps/nowinandroid/feature/topic/navigation/TopicNavigation.kt

```diff
@@ -16,27 +16,27 @@
 
 package com.google.samples.apps.nowinandroid.feature.topic.navigation
 
-import android.net.Uri
 import androidx.annotation.VisibleForTesting
 import androidx.lifecycle.SavedStateHandle
 import androidx.navigation.NavController
 import androidx.navigation.NavGraphBuilder
 import androidx.navigation.NavType
 import androidx.navigation.compose.composable
 import androidx.navigation.navArgument
-import com.google.samples.apps.nowinandroid.core.decoder.StringDecoder
 import com.google.samples.apps.nowinandroid.feature.topic.TopicRoute
+import java.net.URLDecoder
+import java.net.URLEncoder
 
 @VisibleForTesting
 internal const val topicIdArg = "topicId"
 
 internal class TopicArgs(val topicId: String) {
-    constructor(savedStateHandle: SavedStateHandle, stringDecoder: StringDecoder) :
-        this(stringDecoder.decodeString(checkNotNull(savedStateHandle[topicIdArg])))
+    constructor(savedStateHandle: SavedStateHandle) :
+        this(URLDecoder.decode(checkNotNull(savedStateHandle[topicIdArg]), "UTF-8"))
 }
 
 fun NavController.navigateToTopic(topicId: String) {
-    val encodedId = Uri.encode(topicId)
+    val encodedId = URLEncoder.encode(topicId, "UTF-8")
```

Comment:
nit: please move "UTF-8" to a constant

Reply:
Should we really extract these into constants?
Do you prefer a 

```kotlin
private const val UTF_8 = "UTF-8"

URLEncoder.encode(topicId, UTF_8)
```

or directly use Kotlin's already defined `Charsets.UTF_8`?

```kotlin
URLEncoder.encode(topicId, UTF_8.name())
```

Reply:
The constant name should express the meaning of its value, rather than the actual value, so maybe: 

```
private const val URL_CHARACTER_ENCODING  = UTF_8.name()
```

Reply:
Well, if we go the route of expressiveness, I think there is nothing that can beat `URLEncoder.encode(topicId, "UTF-8")`, but OK I'll update that code :)

Reply:
Although, since `.name()` is a function, the `val` can no longer be a `const`.